### ISSUE TITLE
Make exp and iap optional?

### ISSUE THREAD

**Issue by kljensen** [mike-engel/jwt-cli#122]
### Summary

The current behavior of jwt adds `exp` and `iap` claims, e.g.

```
shell>  echo '{"foo": "bar"}'| jwt encode --secret='super-secret-foo' - | jwt decode -                                                                                                  
Token header
------------
{
  "typ": "JWT",
  "alg": "HS256"
}

Token claims
------------
{
  "exp": 1612965158,
  "foo": "bar",
  "iat": 1612963358
}
```

### Steps to reproduce

See above.

### Expected behavior

Each of `exp` and `iat` is "OPTIONAL" in the [JWT spec](https://tools.ietf.org/html/rfc7519#section-4.1.6). I feel the current behavior (addition of `exp` and `iat`) is counterintuitive and that it would be more intuitive if `exp` and `iat` were only added upon request of the user. 

This is, I believe, the most intuitive  behavior for the jwt cli:

|  | `exp` in payload | `exp` not in payload | 
|  ---:         |     :---:      |         :---: |  
| **`--exp` provided** | overwrite `exp` with provided value | set `exp` with provided value | 
| **`--exp` not provided** | keep `exp` from payload | leave `exp` empty |

I think the same behavior for other optional registered claim names (like `iat`) would be more intuitive than the current behavior.

(Thanks for this _fantastic_ tool! 🤘)

**Comment by mike-engel** [comment]
Thanks @kljensen for the suggestion! I agree that `exp` shouldn't be added automatically, so now it will only be added if the `--exp` (or `-e`) flag is used. If no value is provided, it will still default to `+30m`.

As for the `iat` flag, I think that should stay in by default, since it's good for debugging and shouldn't affect any validation. I did add a `--no-iat` flag, however, if you don't want it added.

**Comment by kljensen** [comment]
Thanks Mike! That sounds like a reasonable approach.

**Comment by mike-engel** [comment]
Released with 4.0.0
